src/JsonApiClient.php

```diff
@@ -154,7 +154,8 @@ public function execute(RequestInterface $request, bool $exceptionOnFatalError =
         $response = new HttpResponse(
             $httpResponse->getStatusCode(),
             $httpResponse->getHeaders(),
-            $this->createResponseBody($httpResponse->getBody()->getContents())
+            $this->createResponseBody($httpResponse->getBody()->getContents()),
+            $httpResponse
```

Comment:
What about calling `$httpResponse->getBody()->rewind()` before putting this instance into the client? This ensures the highest compatibility, as the body content still is present.

Reply:
That's a valid objection.